controllers/front/fail.php

```diff
@@ -67,31 +60,28 @@ public function init()
         if (!SaferPayConfig::isVersion17()) {
             return parent::init();
         }
+
         parent::init();
 
         $this->id_cart = (int) Tools::getValue('cartId', 0);
 
         $redirectLink = 'index.php?controller=history';
 
-        $this->id_order = Order::getOrderByCartId((int) $this->id_cart);
         $this->secure_key = Tools::getValue('secureKey');
-        $order = new Order((int) $this->id_order);
 
-        if (!$this->id_order || !$this->module->id || !$this->secure_key || empty($this->secure_key)) {
+        $cart = new Cart($this->id_cart);
+
+        if (!$this->module->id || !$this->secure_key || empty($this->secure_key)) {
```

Comment:
!$this->secure_key || empty($this->secure_key) is duplication, can just check empty